cobra/flux_analysis/find_active_reactions.py

```diff
@@ -0,0 +1,583 @@
+    list
+        List of reaction IDs which can carry flux.
+
+    Notes
+    -----
+    The optmization problem solved is as follow:
+    MILP version:
+    min \sum_{j \in J}{z_pos_j + z_neg_j}
+    s.t.  \sum_{j \in J}{S_ij * v_j} = 0  \forall i \in I
+          LB_j <= v_j <= UB_j             \forall j \in J
+        v_j + \varepsilon * z_pos_j >=  \varepsilon  for j \in J with LB_j >= 0
+        v_j - \varepsilon * z_neg_j <= -\varepsilon  for j \in J with UB_j <= 0
+      v_j + M * z_pos_j >=  \varepsilon  for j \in J with LB_j < 0 and UB_j > 0
+      v_j - M * z_neg_j <= -\varepsilon  for j \in J with LB_j < 0 and UB_j > 0
+      v_j \in \mathbb{R}
+      z_pos_j\in \mathbb{R} for j \in J with LB_j >= 0
+      z_neg_j \in \mathbb{R} for j \in J with UB_j <= 0
+      z_pos_j, z_neg_j \in {0,1} for j \in J with LB_j < 0 and UB_j > 0
+
+    LP version:
+    Solve a number of versions of the LP relaxation of the above problem
+    as follows (cumulative changes in each step):
+    1. Fix all z_pos_j, z_neg_j = 1 for all reversible reactions.
+       Solve the LP to find all active irreversible reactions and
+       some active reversible reactions.
+    2. Fix all z_pos_j, z_neg_j = 1 for all irreversible reactions.
+       Un-fix z_pos_j for the reversible reactions not yet found active.
+       Solve the LP to find some active reversible reactions
+    3. Fix all z_pos_j. Un-fix z_neg_j for reversible reactions not yet found
+       active. Solve the LP to find some active reversible reactions
+    4. Add a randomly weighted min. flux constraint:
+       \sum_{j \in J not yet found active}{w_j * v_j} >= eps
+       Solve and update the set of reversible reactions not yet found active
+       (if any) until infeasibility
+    5. Change the sense and R.H.S. of the min. flux constraint in Step 4 to
+       '<= -eps'. Solve and update until infeasibility
+
+    References
+    ----------
+    Chan, S. H., Wang, L., Dash, S., & Maranas, C. D. (2018). Accelerating flux
+    balance calculations in genome-scale metabolic models by localizing the
+    application of loopless constraints. Bioinformatics, 34(24), 4248-4255.
+
+    """
+
+    solve = solve.lower()
+    if solve not in ["lp", "milp", "fastsnp"]:
+        raise ValueError("Parameter solve must be 'lp', 'milp' or 'fastSNP'.")
+
+    elif solve == "fastsnp":
+        N = fastSNP(model, bigM=bigM, zero_cutoff=zero_cutoff,
+                    relax_bounds=relax_bounds, verbose=verbose)
+        return [model.reactions[j].id for j in range(len(model.reactions))
+                if N[j, :].any()]
+
+    max_bound = max([max(abs(r.upper_bound), abs(r.lower_bound))
+                     for r in model.reactions])
+    if max_bound < float("inf"):
+        bigM = max(bigM, max_bound)
+
+    if relax_bounds and zero_cutoff is None:
+        eps = 1e-5
+    else:
+        eps = normalize_cutoff(model, zero_cutoff)
+
+    eps = max(eps, model.solver.configuration.tolerances.feasibility * 100)
```

Comment:
I think it may still be good to use `model.tolerance' here. Not all tolerances (integrality, feasibility, optimality) are implemented for all solvers (for instance GLPK has no optimality tolerance, they use different measures).